### PR TITLE
Rename download to backupDownload

### PR DESCRIPTION
**Pull request by zwarm**
This PR refactors the downloadable backup feature by renaming the methods/features from **download** to  **backupDownload**. This rename provides a clearer context for future developers. No logic has been changed in the PR.

**Merge instructions**
- Make sure #1772 has been merged
- Remove Not Ready for Merge label
- Merge as normal

**To Test**
The feature is not being used, so validation of CI test will serve the purpose of testing.

### CODE REVIEW

File: example/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/activity/ActivityFixtures.kt [hunk 1]
```diff
@@ -38,7 +38,7 @@ val REWIND_STATUS_RESPONSE = ActivityLogRestClient.RewindStatusResponse(
         can_autoconfigure = true,
         credentials = listOf(),
         rewind = REWIND_RESPONSE)
-val DOWNLOAD_STATUS_RESPONSE = ActivityLogRestClient.DownloadStatusResponse(
+val DOWNLOAD_STATUS_RESPONSE = ActivityLogRestClient.BackupDownloadStatusResponse(
```
Comment (ParaskP7) [inline example/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/activity/ActivityFixtures.kt]: Minor: Most probably the val name should change to include the `BACKUP` keyword as well. Is that right?

File: example/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/activity/ActivityLogRestClientTest.kt [hunk 2]
```diff
@@ -360,55 +360,55 @@ class ActivityLogRestClientTest {
...
+        initPostBackupDownload(rewindId = rewindId, data = response, requestTypes = types)
 
-        val payload = activityRestClient.download(site, rewindId, types)
+        val payload = activityRestClient.backupDownload(site, rewindId, types)
 
         assertEquals(downloadId, payload.downloadId)
     }
 
     @Test
     fun postDownloadOperationError() = test {
```
Comment (ParaskP7) [inline example/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/activity/ActivityLogRestClientTest.kt]: Minor: The test is missing the `backup` keyword as well. Isn't that right?

PS: Same applies for the below test and its private function(s).

File: example/src/test/java/org/wordpress/android/fluxc/store/ActivityLogStoreTest.kt [hunk 3]
```diff
@@ -305,8 +305,8 @@ class ActivityLogStoreTest {
 
     @Test
     fun onDownloadActionCallRestClient() = test {
```
Comment (ParaskP7) [inline example/src/test/java/org/wordpress/android/fluxc/store/ActivityLogStoreTest.kt]: Minor: The test is missing the `backup` keyword as well. Isn't that right?

PS: Same applies for the below test.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt [hunk 4]
```diff
@@ -55,14 +55,14 @@ class ActivityLogStore
                     emitChange(rewind(action.payload as RewindPayload))
                 }
             }
-            DOWNLOAD -> {
+            BACKUP_DOWNLOAD -> {
                 coroutineEngine.launch(AppLog.T.API, this, "ActivityLog: On DOWNLOAD") {
```
Comment (ParaskP7) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt]: Minor: Both download related `On <ACTION>` logs and `emitChang(<functions>)` are missing the `BACKUP` and `backup` keyword respectively. Isn't that right?

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt [hunk 5]
```diff
@@ -157,13 +159,13 @@ class ActivityLogStore
         }
     }
 
-    private fun storeDownloadState(payload: FetchedDownloadStatePayload, action: ActivityLogAction):
+    private fun storeDownloadState(payload: FetchedBackupDownloadStatePayload, action: ActivityLogAction):
```
Comment (ParaskP7) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt]: Minor: This function is missing the `backup` keyword. Isn't that right?

PS: Also, the below `emitDownloadResult(...)` function is missing it.

**Review by ParaskP7 — COMMENTED** [review]
Thanks again for the inclusion in this PR Annmarie! 🌟 

Tiny-minors from my side.

File: example/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/activity/ActivityFixtures.kt (hunk 1 above)
Comment (zwarm) [reply example/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/activity/ActivityFixtures.kt]: Good catch. Thanks

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt [hunk 6]
```diff
@@ -228,16 +230,16 @@ class ActivityLogStore
         val startedAt: String? = null,
         val progress: Int = 0,
         var causeOfChange: ActivityLogAction
-    ) : Store.OnChanged<DownloadError>() {
-        constructor(rewindId: String, error: DownloadError, causeOfChange: ActivityLogAction) :
+    ) : Store.OnChanged<BackupDownloadError>() {
```
Comment (malinajirka) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt:236]: It seems you missed OnDownload -> OnBackupDownload. Wdyt?

**Review by malinajirka — APPROVED** [review]
Thank you @zwarm for making the code clearer for future devs(selfs)! I found one missed spot, other than that it LGTM.

**Pull request merged**